src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -130,14 +130,20 @@ internal CommitManager(ITextView textView, RecentItemsManager recentItemsManager
                 return CommitResultUnhandled;
             }
 
+            var triggerDocument = triggerSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (triggerDocument == null)
+            {
+                return CommitResultUnhandled;
+            }
+
             // Commit with completion service assumes that null is provided is case of invoke. VS provides '\0' in the case.
             char? commitChar = typeChar == '\0' ? null : (char?)typeChar;
             var commitBehavior = Commit(
-                document, completionService, session.TextView, subjectBuffer, 
+                triggerDocument, completionService, session.TextView, subjectBuffer,
```

Comment:
Already signed off, but one thing to add is that maybe we should remove the `document` variable completely so people aren't tempted to use it. Or maybe rename it to disambiguate.

Reply:
Thank you, @dpoeschl !

The `document` variable above is used to get completionService. This service can be used even if we do not have a triggerSnapshot to process `!IsCommitCharacter` and some other cases above. 

Maybe we can get `completionService` more elegant without `document`. However, this would require a more risky fix not valid for a Shiproom Approval.